x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -703,11 +706,12 @@ async function endpointDetailsMiddleware({
       type: 'serverCancelledEndpointListLoading',
     });
   }
-  const { selected_endpoint: selectedEndpoint } = uiQueryParams(getState());
-  if (selectedEndpoint !== undefined) {
-    loadEndpointDetails({ store, coreStart, selectedEndpoint });
+  if (!selectedEndpoint) {
```

Comment:
I got the habit of avoiding checking for undefined like this because sometimes 0 is a valid value and this will discard it. This is of course not the case because the variable is a string but I kept the habit to do it everywhere.

Reply:
ooh yes. That's a good tip. I'll remember that 👍

Reply:
fixed here https://github.com/elastic/kibana/pull/108551/commits/5b9de562ef8f426a807bd2d60cf5317d1a56962d